File: system-tests/log-version-on-startup-test/src/test/java/com/teamscale/client/LogVersionOnStartupTest.java
package com.teamscale.client;

import org.junit.jupiter.api.Test;

import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.regex.Pattern;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Ensures that the agent logs its version on startup at level INFO.
 */
public class LogVersionOnStartupTest {

	private static final Path LOG_DIRECTORY = Paths.get("logTest").resolve("logs");
	private static final String AGENT_VERSION = System.getenv("AGENT_VERSION");

	@Test
	public void systemTest() throws Exception {
		assertThat(Files.exists(LOG_DIRECTORY)).isTrue();
		String logContent = String.join("\n", Files.readAllLines(LOG_DIRECTORY.resolve("teamscale-jacoco-agent.log")));
		assertThat(logContent).containsPattern("INFO.*" + Pattern.quote(AGENT_VERSION));
	}
}

File: system-tests/log-version-on-startup-test/src/test/kotlin/com/teamscale/client/LogVersionOnStartupTest.kt
package com.teamscale.client

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import java.nio.file.Paths
import java.util.regex.Pattern
import kotlin.io.path.exists
import kotlin.io.path.readLines

/**
 * Ensures that the agent logs its version on startup at level INFO.
 */
class LogVersionOnStartupTest {
	@Test
	@Throws(Exception::class)
	fun systemTest() {
		assertTrue(LOG_DIRECTORY.exists())
		val logContent = LOG_DIRECTORY.resolve("teamscale-jacoco-agent.log")
			.readLines()
			.joinToString("\n")
		assertThat(logContent).containsPattern(
			"INFO.*" + Pattern.quote(AGENT_VERSION)
		)
	}

	companion object {
		private val LOG_DIRECTORY = Paths.get("logTest").resolve("logs")
		private val AGENT_VERSION = System.getenv("AGENT_VERSION")
	}
}
